implementation/src/main/java/io/smallrye/config/ConfigMapping.java

```diff
@@ -52,7 +52,21 @@
      */
     NamingStrategy namingStrategy() default NamingStrategy.KEBAB_CASE;
 
-    enum NamingStrategy {
+    /**
+     * Match config mapping method names and configuration properties using bean-style getter names. This removes the
+     * prefixes <code>get</code> or <code>is</code> and decapitalizes the next character. By default, bean-style getter
+     * matching is <code>disabled</code>.
+     * <p>
+     * Config mapping declarations should prefer to use simple names that match one-to-one with their configuration
+     * names. Bean-style getter matching allows multiple method names to match the same configuration name. For
+     * instance, <code>getFoo</code> and <code>isFoo</code> both match <code>foo</code>, which may not be intended.
```

Comment:
Do we know which one Kotlin uses? Maybe we should have this be an `enum` with each possiblity?

Reply:
From what I can tell is `get`, but you can override the name in Kotlin.

Reply:
@dmlloyd are you ok to merge as is? Or do you prefer something different?

Reply:
I had a thought to differentiate between these two cases, but I think it probably doesn't matter too much.